### PR TITLE
build(aio): add grouping of API overloads

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: aio/src/styles/2-modules/_details.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+summary {
+  display: block;
+  font-size: 18px;
+  position: relative;
+  padding: 16px 23px;
+  margin: 0;
+  background-color: $mist;
+  color: #373E41;
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_details.scss]: Hard-coded color 😱

File: aio/src/styles/2-modules/_details.scss [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_details.scss]: Maybe also `cursor: pointer`?

File: aio/tools/transforms/templates/api/lib/memberHelpers.html [hunk 3]
```diff
@@ -9,20 +9,20 @@
...
+{%- macro renderMember(member, truncate) -%}
     {%- if member.accessibility !== 'public' %}{$ member.accessibility $} {% endif -%}
     {%- if member.isGetAccessor %}get {% endif -%}
     {%- if member.isSetAccessor %}set {% endif -%}
     {%- if member.isStatic  %}static {% endif -%}
-    {$ member.name $}{$ member.typeParameters | escape $}{$ params.paramList(member.parameters) | trim $}
+    {$ member.name $}{$ member.typeParameters | escape $}{$ params.paramList(member.parameters, truncate) | trim $}
     {%- if member.isOptional  %}?{% endif -%}
-    {$ params.returnType(member.type) $}
+    {$ params.returnType(member.type) | trim | truncateFirstLine $}
```
Comment (gkalpak) [inline aio/tools/transforms/templates/api/lib/memberHelpers.html]: Should `truncateFirstLine` be conditional?

**Review by gkalpak — COMMENTED** [review]
Looks really nice.

Nit:
One thing that threw me off is truncating a multi-line object that also happens to be the last item before a closing `)`:

```ts
foo(bar: {
  baz: qux
})

// ...becomes...

foo(bar: {...)
```

That `{...)` looks kind of weird. It would be awesome if `truncateFirstLine` could detect that `{` is the last char and put a closing `}` at the end.

File: aio/src/styles/2-modules/_details.scss (hunk 1 above)
Comment (petebacondarwin) [inline aio/src/styles/2-modules/_details.scss]: I copied this from the resources page styling :-P

**Comment by petebacondarwin** [comment]
@gkalpak I fixed the truncating to close any open brackets.

**Comment by petebacondarwin** [comment]
I quietened the styling of the overloads, so that they do not look like a new item.
I have also taken the opportunity to tidy up the API doc layout and styling.
Please do take a look when the preview is ready.

**Comment by petebacondarwin** [comment]
I just realised that we need to be more clever with overloads if the doc type is an interface or abstract class. See https://pr18081-6a21073.ngbuilds.io/api/animations/AnimationBuilder#build

**Review by IgorMinar — APPROVED** [review]

File: aio/tools/transforms/angular-base-package/rendering/truncateCode.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
+module.exports = function() {
+  return {
+    name: 'truncateCode',
+    process: function(str, lines) {
+      if (lines === undefined) return str;
+
+      const parts = str && str.split && str.split(/\r?\n/);
+      if (parts && parts.length > lines) {
+        return balance(parts[0] + '...', ['{', '(', '['], ['}', ')', ']']);
```
Comment (gkalpak) [inline aio/tools/transforms/angular-base-package/rendering/truncateCode.js]: ❤️

**Pull request merged**